snuba/datasets/__init__.py

```diff
@@ -32,3 +32,54 @@ def row_from_processed_message(self, message):
             values.append(value)
 
         return values
+
+    # These method should be removed once we will have dataset specific query processing in
+    # the dataset class instead of util.py and when the dataset specific logic for processing
+    # Kafka messages will be in the dataset as well.
+
+    def get_metadata_columns(self):
```

Comment:
All these methods should move to the events dataset class instead of being in the parent class.
But before doing that we need to have the event specific logic from processor and util to move into an event specific class. so the methods will not be here at all.

Reply:
I agree with this in general (I am interpreting "these methods" as all of the methods in this added chunk, not just the first set that are not fully implemented) based on our discussion yesterday about this functionality being events-specific.

I don't understand why `get_metadata_columns`, `get_promoted_tag_columns`, etc. are defined in this class, but don't have a body implemented, though. Is this just because they are called by `get_promoted_cols`, etc? If that's the case, shouldn't these raise `NotImplementedError` rather than returning `None`?

Also, as a nit: it drives me a little crazy to have several methods use the full term "columns" but then also have "cols" in `get_promoted_cols`.

Reply:
So, these methods do not make sense for datasets other than events, though they are still called by util.py. 
As long as we have one dataset only all good. If we introduce a second one before refactoring util.py we will have to either provide a fake implementation for that dataset or to have a base implementation here.
Thinking over it returning None is not a valid basic implementation, so we are screwed anyway.
Will change into NotImplemented.

Also about "cols", yes that is a good point. Will change to columns. That was the result of a codemod

Reply:
Makes sense, thanks for the clarification/changes. 👍